demo/app/lib/views/handle_redirect_uri.dart

```diff
@@ -67,56 +70,63 @@ class HandleRedirectUriState extends State<HandleRedirectUri> {
     return didID;
   }
 
-  Future<Widget>? launchCredPreview() async {
+  launchCredPreview() async {
     final SharedPreferences pref = await prefs;
     await _createDid();
-    pref.setString('userDID',userDIDId);
-    pref.setString('userDIDDoc',userDIDDoc);
+    pref.setString('userDID', userDIDId);
+    pref.setString('userDIDDoc', userDIDDoc);
     if (_redirectUri != null) {
       try {
-        var credentials = await WalletSDKPlugin.requestCredentialWithAuth(_redirectUri.toString());
-        String? issuerURI = await WalletSDKPlugin.issuerURI();
-        var serializedDisplayData = await WalletSDKPlugin.serializeDisplayData([credentials], issuerURI!);
-        log("serializedDisplayData -> $serializedDisplayData");
-        var activities = await WalletSDKPlugin.storeActivityLogger();
-        var credID = await WalletSDKPlugin.getCredID([credentials]);
-        await _storageService.addActivities(ActivityDataObj(credID!, activities));
-        pref.setString("credID", credID);
-         _navigateToCredPreviewScreen(credentials, issuerURI, serializedDisplayData!, userDIDId, credID);
-      } catch (error) {
-        if (!error.toString().contains("UKN2-000")){
-          Navigator.push(
-              context,
-              MaterialPageRoute(
-                  builder: (context) =>
-                      CustomError(requestErrorTitleMsg: "Redirect uri error", requestErrorSubTitleMsg: error.toString())));
+        if (widget.flowType == "issuer-initiated-flow") {
+          var credentials = await WalletSDKPlugin.requestCredentialWithAuth(_redirectUri.toString());
+          var issuerURI = await WalletSDKPlugin.issuerURI();
+          var serializedDisplayData = await WalletSDKPlugin.serializeDisplayData([credentials], issuerURI!);
+          var activities = await WalletSDKPlugin.storeActivityLogger();
+          var credID = await WalletSDKPlugin.getCredID([credentials]);
+          await _storageService.addActivities(ActivityDataObj(credID!, activities));
+          pref.setString("credID", credID);
+          _navigateToCredPreviewScreen(credentials, issuerURI, serializedDisplayData!, userDIDId, credID);
         } else {
+          var credentials = await WalletSDKPlugin.requestCredentialWithWalletInitiatedFlow(_redirectUri.toString());
+          var issuerURI = widget.issuerURI;
+          var serializedDisplayData = await WalletSDKPlugin.serializeDisplayData([credentials], issuerURI!);
+          // TODO: Issue-518 Add activity logger support for wallet-initiated-flow
+          _navigateToCredPreviewScreen(credentials, issuerURI, serializedDisplayData!, userDIDId, "");
+        }
+      } catch (error) {
+        log(error.toString());
+        if (error.toString().contains("UKN2-000") || error.toString().contains("OCI1-0008")) {
```

Comment:
Shouldn't ever check for `UKN2-000`. It's a catch-all type of error for errors that don't have defined error codes.

Reply:
Also, check for the actual error category instead of the full code. We don't currently guarantee that the error code will stay the same.

Reply:
Also, why the specific check for `OCI1-0008`?

Reply:
I think you probably should trigger this condition on any error that happens.

Reply:
@DRK3  There are multiple error that comes from sdk while we are waiting for request credential to issue the credential. I noticed it takes few second and then, like data not found etc comes from sdk while we are waiting for call to complete so, I will create a separate issue to look into more detail into this.